src/ui/src/core_components/other/CoreJsonViewer.vue

```diff
@@ -0,0 +1,130 @@
+import { JsonViewerTogglePayload } from "../../wds/WdsJsonViewer.utils";
+import { FieldCategory, FieldType } from "../../writerTypes";
+
+const description = "A standalone element to view a JSON data.";
+
+const initialData = {
+	name: "JSON Viewer",
+	description: "A JSON tree viewer were you can expand key",
+	sample: {
+		description: "This sample is opened by default",
+		bool: true,
+		null: null,
+		list: [1, "two", { key: 3 }],
+	},
+	sampleClosed: {
+		description: "This sample is not opened by default",
+	},
+	createdAt: new Date(),
+};
+
+const toggleHandlerStub = `
+def handle_json_tree_toggle(state, payload):
+
+	# string array representing the keys of the path
+	path = payload.get("path")
+
+
+	# boolean if the path was open (True) or closed (False)
+	open = payload.get("open")
+
+	# ...`;
+
+export default {
+	writer: {
+		name: "JSON Viewer",
+		description,
+		category: "Other",
+
+		events: {
+			"wf-toggle": {
+				desc: "Capture JSON tree element open.",
+				stub: toggleHandlerStub.trim(),
+			},
+		},
+
+		fields: {
+			data: {
+				name: "Data",
+				init: JSON.stringify(initialData),
+				type: FieldType.Text,
+			},
+			pathsOpen: {
+				name: "Paths open",
+				init: JSON.stringify({ "sample.list.2": true }),
+				type: FieldType.KeyValue,
+			},
+			collapseAngleColor: {
+				name: "Collapse angle",
+				type: FieldType.Color,
+				category: FieldCategory.Style,
+				applyStyleVariable: true,
+			},
+			jsonViewerValueColor: {
+				name: "JSON value",
+				type: FieldType.Color,
+				category: FieldCategory.Style,
+				applyStyleVariable: true,
+			},
+			jsonViewerIndentationSpacing: {
+				name: "JSON indentation",
+				type: FieldType.Width,
+				category: FieldCategory.Style,
+				applyStyleVariable: true,
+			},
+			separatorColor,
+			cssClasses,
+		},
+		previewField: "data",
+	},
+};
+</script>
+
+<script setup lang="ts">
+import { ComponentPublicInstance, computed, inject, ref } from "vue";
+import injectionKeys from "../../injectionKeys";
+import WdsJsonViewer from "../../wds/WdsJsonViewer.vue";
+import WdsJsonViewerChildrenCounter from "../../wds/WdsJsonViewerChildrenCounter.vue";
+import { useJSONParser } from "./useJSONParser";
+
+const rootInstance = ref<ComponentPublicInstance | null>(null);
+
+const fields = inject(injectionKeys.evaluatedFields);
+
+const pathsOpen = computed<string[][]>(() => {
+	if (!fields.pathsOpen.value) return [];
+	return Object.entries(fields.pathsOpen.value as Record<string, unknown>)
+		.filter(([_, value]) => value === true || value === "true")
+		.map(([key]) => key.split("."));
+});
+
+const { data, error } = useJSONParser(fields.data);
+
+function onToggle(payload: JsonViewerTogglePayload) {
+	if (!rootInstance.value) return;
+	const event = new CustomEvent("wf-toggle", { detail: payload });
+	rootInstance.value.$el.dispatchEvent(event);
+}
+</script>
+
+<style scoped>
+.error {
+	color: #ff5d00;
```

Comment:
For error we use a #FFCFC2 background and primary text color for the text. Designers prefer to avoid colored text.

For now you can hardcode #FFCFC2 but I'm guessing we'll need a setting somehow.

Reply:
Ha, I copied it from here

https://github.com/madeindjs/writer-framework/blob/c7f131c4b3a2e08c5c9c1ee263b3f5e2985bf837/src/ui/src/builder/BuilderEditor.vue#L447-L449

I updated it